docs/dev/ffi/when-to-use-what-in-the-ffi.md

```diff
@@ -56,27 +35,26 @@ structures containing them.
 
 ### Primitives
 
-Specifically numeric primitives. These we'll tackle first since they're the
-easiest.
+Numeric primitives are the easiest to pass between languages.
+The main recommendation is: use the equivalent and same-sized type as the one provided by Rust.
 
-In general, you can just pass them as you wish. There are a couple of
-exceptions/caveats. All of them are caused by JNA/Android issues (Swift has very
-good support for calling over the FFI), but it's our lowest common denominator.
+There are a couple of exceptions/caveats, especially for Kotlin. All of them are caused by JNA/Android issues.
+Swift has very good support for calling over the FFI.
 
 1. `bool`: Don't use it. JNA doesn't handle it well. Instead, use a numeric type
-    (like `u8`) and represent 0 for false and 1 for true for interchange over the
-    FFI, converting back to a Kotlin `Boolean` or swift `Bool` after (as to
+    (like `u8`) and represent 0 for `false` and 1 for `true` for interchange over the
+    FFI, converting back to Kotlin's `Boolean` or Swift's `Bool` after (as to
     not expose this somewhat annoying limitation in our public API).
+    All wrappers already include utility functions to turn 8-bit integers (`u8`) back to booleans
+    (`toBool()` or equivalent methods).
 
 2. `usize`/`isize`: These cause the structure size to be different based on the
-   platform. JNA does handle this if you use `NativeSize`, but it's awkward,
-   incompatible with it's Direct Mapping optimization (which we don't use but
-   want to in the future), and has more overhead than just using `i64`/`i32` for
-   `Long`/`Int`. (You can also use `u64`/`u32` for `Long`/`Int`, if you're certain the
-   value is not negative)
+   platform. JNA does handle this if you use `NativeSize`, but it's awkward.
+   Use the size-defined integers instead, such as `i64`/`i32` and their language-equivalents
+   (Kotlin: `Long`/`Int`, Swift:`UInt64`/`UInt32`).
+   *Caution:* In Kotlin integers are signed by default. You can use `u64`/`u32` for `Long`/`Int`, if you're certain the value is not negative.
```

Comment:
Ugh. I don't like "if you're certain". Programmers make mistakes (or maybe that's just me) so I'd prefer some sort of "You _can_ do this if you ensure the values are non-negative through asserts or error reporting code" or something

Reply:
will change. I do agree, but for now that's the state we're in :/

Reply:
```suggestion
   *Caution:* In Kotlin integers are signed by default. You can use `u64`/`u32` for `Long`/`Int` if you ensure the values are non-negative through asserts or error reporting code.
```